freqtrade/resolvers/iresolver.py

```diff
@@ -0,0 +1,68 @@
+# pragma pylint: disable=attribute-defined-outside-init
+
+"""
+This module load custom objects
+"""
+import importlib.util
+import inspect
+import logging
+from pathlib import Path
+from typing import Optional, Dict, Type, Any
+
+logger = logging.getLogger(__name__)
+
+
+class IResolver(object):
+    """
+    This class contains all the logic to load custom hyperopt class
+    """
+
+    def __init__(self, object_type, config: Optional[Dict] = None) -> None:
+        """
+        Load the custom class from config parameter
+        :param config: configuration dictionary or None
+        """
+        config = config or {}
+
+    @staticmethod
+    def _get_valid_objects(object_type, module_path: Path,
+                           object_name: str) -> Optional[Type[Any]]:
+        """
+        Returns a list of all possible objects for the given module_path of type oject_type
+        :param object_type: object_type (class)
+        :param module_path: absolute path to the module
+        :param object_name: Class name of the object
+        :return: Tuple with (name, class) or None
+        """
+
+        # Generate spec based on absolute path
+        spec = importlib.util.spec_from_file_location('unknown', str(module_path))
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)  # type: ignore # importlib does not use typehints
+
+        valid_objects_gen = (
+            obj for name, obj in inspect.getmembers(module, inspect.isclass)
+            if object_name == name and object_type in obj.__bases__
+        )
+        return next(valid_objects_gen, None)
```

Comment:
next(...) will return the next item from the iterator but the parent function name is _get_valid_objects. just thinking loud ...

Reply:
you're right, the function name is misleading (i think a long time ago it did something else ...) 
I'll try to rename it to something more appropriate